### DISCUSSION TITLE
/posts/tmux-setup

### DISCUSSION THREAD

**Discussion by giscus** [josean-dev/sveltekit-blog#5] [General]
# /posts/tmux-setup

Follow along my youtube video with this blog post to get up and running with Tmux quickly and use it alongside Neovim for an awesome software development workflow.

https://www.josean.com/posts/tmux-setup

**Comment by MaximeCap** [comment]
Hey I have quite the same setup as you in nvim and tmux but I ran into an issue where Tmux take the lead of <C-j|k> instead of nvim when by exemple Fuzysearch or auto complete. Do you have any tips on that or did you faced that problem already ?

**Reply by rendivs925** [reply]
cool

**Reply by rendivs925** [reply]
tes

**Comment by rendivs925** [comment]
this is super cool sir

**Comment by rendivs925** [comment]


**Reply by rendivs925** [reply]
i am just triying your comment feature

**Comment by rendivs925** [comment]
tes

**Comment by rendivs925** [comment]
tes

**Comment by wilbrijo** [comment]
Josean,
Thanks for another awesome tutorial.  I had a question - installed the continuum and resurrect plugins, but they're whigging out on me.  I've tried killing sessions, and no matter what I do, it will always restore the old sessions when I start tmux again - this is often unhelpful.  Is there a way I can keep the feature on in case I close out/shutdown unintentionally, but also definitively kill a session when I want to?

**Reply by lovelindhoni** [reply]
hey willbrijo, did u figure how to fix this one. Its whigging me out too man 😭

**Reply by JoshdfG** [reply]
@wilbrijo @lovelindhoni  its almost a year now i hope you've figured it out, but if not, you can resolve it by **using ctrl + a which he used or whatever you are using own your end then ctrl + s to save your new session so it keeps the new panes**
This would basically update session and you stop getting the old ones.

**Comment by nightfury99** [comment]
Josean, thank you for your tutorial, I do want to ask one question. How to switch between different color for tmux-themepack. For example:

```
set -g @themepack "powerline/default/cyan"
// change to 
set -g @themepack "powerline/default/green"
```

and (CTRL + a + I) also does not work 😓

**Reply by Fah42** [reply]
did you install tmp ?

git clone https://github.com/tmux-plugins/tpm ~/.tmux/plugins/tpm

**Reply by nightfury99** [reply]
Yes I did install tpm. Right now im using cyan theme and want to change it to green. Set it to green after using cyan does not taking any effect. Maybe need to restart/refresh tmux.

**Reply by nightfury99** [reply]
apparently we need to kill all existing sessions and it will reload the conf after creating new session.

**Comment by vedanta28** [comment]
For people having trouble with Ctrl [HJKL]. Add these lines in the .tmux.conf file.
```
# Set key bindings to switch panes
bind -n C-h select-pane -L
bind -n C-j select-pane -D
bind -n C-k select-pane -U
bind -n C-l select-pane -R

# Ensure tmux recognizes the control key for pane switching
bind-key C-h if-shell "[[ $(tmux display -p '#{pane_in_mode}') -eq 1 ]]" "send-keys C-h" "select-pane -L"
bind-key C-j if-shell "[[ $(tmux display -p '#{pane_in_mode}') -eq 1 ]]" "send-keys C-j" "select-pane -D"
bind-key C-k if-shell "[[ $(tmux display -p '#{pane_in_mode}') -eq 1 ]]" "send-keys C-k" "select-pane -U"
bind-key C-l if-shell "[[ $(tmux display -p '#{pane_in_mode}') -eq 1 ]]" "send-keys C-l" "select-pane -R"
```

**Reply by vedanta28** [reply]
```
is_vim="ps -o state= -o comm= -t '#{pane_tty}' | grep -iqE '^[^TXZ ]+ +(\\S+\\/)?g?\.?(view|n?vim?x?)(-wrapped)?(diff)?$'"

bind-key -n 'C-h' if-shell "$is_vim" 'send-keys C-h' { if -F '#{pane_at_left}' '' 'select-pane -L' }
bind-key -n 'C-j' if-shell "$is_vim" 'send-keys C-j' { if -F '#{pane_at_bottom}' '' 'select-pane -D' }
bind-key -n 'C-k' if-shell "$is_vim" 'send-keys C-k' { if -F '#{pane_at_top}' '' 'select-pane -U' }
bind-key -n 'C-l' if-shell "$is_vim" 'send-keys C-l' { if -F '#{pane_at_right}' '' 'select-pane -R' }

bind-key -T copy-mode-vi 'C-h' if -F '#{pane_at_left}' '' 'select-pane -L'
bind-key -T copy-mode-vi 'C-j' if -F '#{pane_at_bottom}' '' 'select-pane -D'
bind-key -T copy-mode-vi 'C-k' if -F '#{pane_at_top}' '' 'select-pane -U'
bind-key -T copy-mode-vi 'C-l' if -F '#{pane_at_right}' '' 'select-pane -R'
```

**Comment by aayu-11** [comment]
hey I'm struggling to configure the status bar in my tmux window how can i configure that pls help

**Reply by themellowpea** [reply]
make sure you are installing the 'themepack' plugin: 

`set -g @plugin 'jimeh/tmux-themepack'`

from the [tmux docs](https://github.com/jimeh/tmux-themepack)

**Comment by Lrodriguez2027** [comment]
Hey guys, 

I have a question. I just finished this tutorial and before this I did the neovim setup (2024) from Jose. Is there a way to set up tmux within neovim. Now I have to start a session with tmux before opening neovim and I would like to start tmux when I start neovim. Please keep in mind im a noob as I just started following Josean's vidoes.

**Comment by Straven** [comment]
I followed all the steps but I can't get tmux binds to work even default ones. I can't understand what am I doing wrong.

**Reply by mr-zlaam** [reply]
copy paste this into your tmux every thing will work out of the box

```
set -g default-terminal "screen-256color"

set -g prefix C-w
unbind C-b
bind-key C-w send-prefix

unbind %
bind | split-window -h

unbind '"'
bind - split-window -v

unbind r
bind r source-file ~/.tmux.conf
# for moving between pans
bind -r j resize-pane -D 5
bind -r k resize-pane -U 5
bind -r l resize-pane -R 5
bind -r h resize-pane -L 5

# Ensure tmux recognizes the control key for pane switching
bind-key C-h if-shell "[[ $(tmux display -p '#{pane_in_mode}') -eq 1 ]]" "send-keys C-h" "select-pane -L"
bind-key C-j if-shell "[[ $(tmux display -p '#{pane_in_mode}') -eq 1 ]]" "send-keys C-j" "select-pane -D"
bind-key C-k if-shell "[[ $(tmux display -p '#{pane_in_mode}') -eq 1 ]]" "send-keys C-k" "select-pane -U"
bind-key C-l if-shell "[[ $(tmux display -p '#{pane_in_mode}') -eq 1 ]]" "send-keys C-l" "select-pane -R"

# For resizing pan
bind -r m resize-pane -Z

set -g mouse on

set-window-option -g mode-keys vi

bind-key -T copy-mode-vi 'v' send -X begin-selection # start selecting text with "v"
bind-key -T copy-mode-vi 'y' send -X copy-selection # copy text with "y"

unbind -T copy-mode-vi MouseDragEnd1Pane # don't exit copy mode after dragging with mouse
# disable bottom line
set-option -g status off
# tpm plugin
set -g @plugin 'tmux-plugins/tpm'

# list of tmux plugins
set -g @plugin 'christoomey/vim-tmux-navigator' # for navigating panes and vim/nvim with Ctrl-hjkl
set -g @plugin 'jimeh/tmux-themepack' # to configure tmux theme
#set -g @plugin 'tmux-plugins/tmux-resurrect' # persist tmux sessions after computer restart
set -g @plugin 'tmux-plugins/tmux-continuum' # automatically saves sessions for you every 15 minutes

set -g @themepack 'powerline/default/cyan' # use this theme for tmux

set -g @resurrect-capture-pane-contents 'on' # allow tmux-ressurect to capture pane contents
set -g @continuum-restore 'on' # enable tmux-continuum functionality

# Initialize TMUX plugin manager (keep this line at the very bottom of tmux.conf)
run '~/.tmux/plugins/tpm/tpm'

```

**Reply by mr-zlaam** [reply]
but my binding key is ctrl+w instead of ctrl+a
